### ISSUE TITLE
Prevent async `describe`

### ISSUE THREAD

**Issue by louy** [jest-community/eslint-plugin-jest#18]
Thanks for this great plugin!

A mistake I've seen some people make is using an async function for `describe`, which then makes jest fail silently for a case like the following:

```js
describe('sample case', () => {
  it('works', () => {
    expect(true).toEqual(true);
  });
  describe('async', async () => {
    await new Promise(setImmediate);
    it('breaks', () => {
      throw new Error('Fail');
    });
  });
});
```

It would be nice to have a rule that prevents this. Same would go for returning a promise (or any other value) inside `describe`. 

```js
describe('sample case', () => {
  it('works', () => {
    expect(true).toEqual(true);
  });
  describe('async', () => {
    return Promise.resolve().then(() => {
      it('breaks', () => {
        throw new Error('Fail');
      });
    });
  });
});
```

What do you guys think?

**Comment by SimenB** [comment]
I think this makes sense. We should also check for a `done` callback in `describe`-blocks.

(Anecdotally the first eslint rule I ever wrote: https://github.com/tlvince/eslint-plugin-jasmine/blob/master/lib/rules/no-suite-callback-args.js)

**Comment by SimenB** [comment]
Also of course check for a return value within `describe`, and if it's `async`

**Comment by macklinu** [comment]
I'm interested in contributing this feature - will open a PR for discussion on a WIP implementation.
